templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -340,84 +316,115 @@ contents:
-        echo "WARN: OVS did not succesfully activate NM connection. Attempting to bring up connections"
-        counter=0
-        while [ $counter -lt 5 ]; do
-          if nmcli conn up "$ovs_interface"; then
-            echo "OVS successfully configured"
-            update_nm_conn_files "$bridge_name" "$port_name"
-            copy_nm_conn_files
-            configure_driver_options ${iface}
-            configured=true
-            break
-          fi
-          sleep 5
-          counter=$((counter+1))
-        done
-      fi
-      if [ $configured != true ]; then
-        echo "ERROR: Failed to activate $ovs_interface NM connection"
-        exit 1
-      fi
+      configure_driver_options "${iface}"
+      update_nm_conn_files "$bridge_name" "$port_name"
+      persist_nm_conn_files
     }
 
+    # Used to remove a bridge
     remove_ovn_bridges() {
       bridge_name=${1}
       port_name=${2}
-      iface=""
-      if nmcli connection show ovs-port-${port_name} &> /dev/null; then
-        iface=$(nmcli --get-values connection.interface-name connection show ovs-port-${port_name})
-      fi
 
       # Reload configuration, after reload the preferred connection profile
       # should be auto-activated
       update_nm_conn_files ${bridge_name} ${port_name}
       rm_nm_conn_files
-      nmcli c reload
-      sleep 5
 
       # NetworkManager will not remove ${bridge_name} if it has the patch port created by ovn-kubernetes
       # so remove explicitly
       ovs-vsctl --timeout=30 --if-exists del-br ${bridge_name}
-
-      # In some cases the preferred connection profile is not auto-activated
-      # (maybe due to differences in NM versions) so try to activate it
-      # explicitly.
-      if [ -n "$iface" ]; then
-        nmcli device connect $iface
-      fi
     }
 
-    # removes any previous ovs configuration
+    # Removes any previous ovs configuration
     rollback() {
+      echo "Reverting any previous OVS configuration"
+      
       remove_ovn_bridges br-ex phys0
       if [ -d "/sys/class/net/br-ex1" ]; then
         remove_ovn_bridges br-ex1 phys1
       fi
-      ovs-vsctl --timeout=30 --if-exists del-br br-int -- --if-exists del-br br-local
-
+      
       echo "OVS configuration successfully reverted"
     }
 
+    # Activates a NM connection profile
+    activate_nm_conn() {
+      local conn="$1"
+      for i in {1..10}; do
+        echo "Attempt $i to bring up connection $conn"
+        nmcli conn up "$conn" && s=0 && break || s=$?
+        sleep 5
+      done
+      if [ $s -eq 0 ]; then
+        echo "Brought up connection $conn successfully"
+      else
+        echo "ERROR: Cannot bring up connection $conn after $i attempts"
+      fi
+      return $s
+    }
+
+    # Reloads NetworkManager
+    reload_nm() {
+      echo "Reloading NetworkManager..."
+
+      # set network off, so that autoconnect priority is evaluated when turning
+      # it back on
+      nmcli network off
```

Comment:
I think if you do this then we wont be able to run the script remotely anymore. We would lose IP connectivity to the node if running in a shell. Not sure if it matters or not, just pointing it out.

Reply:
It looks to run fine remotely for me as long as connectivity is restored in a timely fashion. I guess this is the same consideration that we have as we move interfaces around and reconfigure things.